sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -0,0 +1,241 @@
+
+        public UcfgInstructionFactory(SemanticModel semanticModel, UcfgObjectFactory objectFactory)
+        {
+            this.semanticModel = semanticModel;
+            this.objectFactory = objectFactory;
+        }
+
+        public Instruction Create(SyntaxNode syntaxNode)
+        {
+            switch (syntaxNode)
+            {
+                case ObjectCreationExpressionSyntax objectCreation:
+                    return CreateFromObjectCreationExpression(objectCreation);
+
+                case IdentifierNameSyntax identifierName:
+                    return CreateFromIdentifierName(identifierName);
+
+                case VariableDeclaratorSyntax variableDeclarator:
+                    return CreateFromVariableDeclarator(variableDeclarator);
+
+                case BinaryExpressionSyntax binaryExpression:
+                    return CreateFromBinaryExpression(binaryExpression);
+
+                case InvocationExpressionSyntax invocationExpression:
+                    return CreateFromInvocationExpression(invocationExpression);
+
+                case AssignmentExpressionSyntax assignmentExpression:
+                    return CreateFromAssignmentExpression(assignmentExpression);
+
+                case BaseMethodDeclarationSyntax methodDeclaration:
+                    return CreateEntryPointInstruction(methodDeclaration);
+
+                default:
+                    return DefaultCreate(syntaxNode);
+            }
+        }
+
+        private Instruction DefaultCreate(SyntaxNode node) =>
+            objectFactory.CreateConstant(node);
+
+        private Instruction CreateFromObjectCreationExpression(ObjectCreationExpressionSyntax objectCreationExpression)
+        {
+            var ctorSymbol = GetSymbol(objectCreationExpression) as IMethodSymbol;
+            if (ctorSymbol == null)
+            {
+                return objectFactory.CreateConstant(objectCreationExpression);
+            }
+
+            var arguments = BuildArguments(objectCreationExpression, ctorSymbol).ToArray();
+
+            return AddMethodCall(objectCreationExpression, ctorSymbol, arguments);
+        }
+
+        private Instruction CreateFromIdentifierName(IdentifierNameSyntax identifierName)
+        {
+            var identifierSymbol = GetSymbol(identifierName);
+
+            if (identifierSymbol is IPropertySymbol property)
+            {
+                return AddMethodCall(identifierName, property.GetMethod);
+            }
+            else if (IsLocalVarOrParameterOfTypeString(identifierSymbol))
+            {
+                return objectFactory.CreateVariable(identifierName, identifierName.Identifier.Text);
+            }
+            else
+            {
+                return objectFactory.CreateConstant(identifierName);
+            }
+        }
+
+        private Instruction CreateFromVariableDeclarator(VariableDeclaratorSyntax variableDeclarator)
+        {
+            if (variableDeclarator.Initializer != null)
```

Comment:
[minor] You could exit early instead of having this nesting.

Reply:
I'll have to put two `return null` statements if I do that.